src/pubsub/ua_pubsub_config.c

```diff
@@ -0,0 +1,1693 @@
+            statusCode = UA_STATUSCODE_BADINTERNALERROR;
+            break;
+        }
+
+        filePointer = fopen(fileNameChar, "rb");
+        if(filePointer == NULL) {
+            UA_LOG_ERROR(UA_Log_Stdout, UA_LOGCATEGORY_SERVER, 
+                         "[UA_readFileToBuffer] Opening file \"%s\" failed (file does not exist)", fileNameChar);
+            statusCode = UA_STATUSCODE_BADINTERNALERROR; 
+            break;
+        }
+
+        /* Determine file size */
+        int ret = fseek(filePointer, 0, SEEK_END);
+        if(ret < 0) {
+            statusCode = UA_STATUSCODE_BADINTERNALERROR; 
+            break;
+        }
+        long int fileSize = ftell(filePointer);
+        if(fileSize < 0) {
+            statusCode = UA_STATUSCODE_BADINTERNALERROR; 
+            break;
+        }
+        ret = fseek(filePointer, 0, SEEK_SET);
+        if(ret < 0) {
+            statusCode = UA_STATUSCODE_BADINTERNALERROR; 
+            break;
+        }
+
+        buffer->data = (UA_Byte*)UA_calloc((size_t)fileSize, sizeof(UA_Byte));
+        if(buffer->data == NULL) {
+            UA_LOG_ERROR(UA_Log_Stdout, UA_LOGCATEGORY_SERVER, "[UA_readFileToBuffer] Allocation of buffer failed!");
+            statusCode = UA_STATUSCODE_BADOUTOFMEMORY; 
+            break;
+        }
+        buffer->length = (size_t)fileSize;
+
+        size_t bytes = fread(buffer->data, sizeof(UA_Byte), buffer->length, filePointer);
+        if(bytes == 0) {
+            UA_LOG_ERROR(UA_Log_Stdout, UA_LOGCATEGORY_SERVER, 
+                        "[UA_readFileToBuffer] Reading file %s into buffer failed!", fileNameChar);
+            free(buffer->data);
+            *buffer = UA_STRING_NULL;
+            statusCode = UA_STATUSCODE_BADINTERNALERROR; 
+            break;
+        }
+    } while(0);
+
+    if(filePointer != NULL) {
+        fclose(filePointer);
+    }
+    if(fileNameChar != NULL) {
+        free(fileNameChar);
+        fileNameChar = NULL;
+    }
+
+    return statusCode;
+}
+
+/* UA_decodeBinFile() */
+/**
+ *  @brief      Decodes buffered binary config file
+ * 
+ *  @param      buffer      [in]    Object that buffers the bin file
+ *  @param      offset      [bi]    Offset in the buffer
+ *  @param      dst         [out]   Object that shall contain the decoded data
+ * 
+ *  @return     UA_StatusCode
+ */
+UA_StatusCode 
+UA_decodeBinFile(const UA_ByteString *buffer, size_t offset, UA_ExtensionObject *dst)
+{
+    UA_StatusCode statusCode = UA_STATUSCODE_GOOD;
+    if(buffer == NULL || dst == NULL) {
+        UA_LOG_ERROR(UA_Log_Stdout, UA_LOGCATEGORY_SERVER, "[UA_decodeBinFile] invalid argument");
+        statusCode = UA_STATUSCODE_BADINVALIDARGUMENT;
+    } else {
+        statusCode = UA_ExtensionObject_decodeBinary(buffer, &offset, dst);
+        if(statusCode != UA_STATUSCODE_GOOD) {
+            UA_LOG_ERROR(UA_Log_Stdout, UA_LOGCATEGORY_SERVER, "[UA_decodeBinFile] decoding UA_Binary failed");
+        }
+    }
+    return statusCode;
+}
```

Comment:
This function looks generic and not pubsub config specific. Therefore, I’m not sure if we really should make it part of the API.